### PR TITLE
Added `Get-FileEncoding` cmdlet

### LINKED ISSUE #2290: Add Get-FileEncoding cmdlet or function.
This is common task I see across many PowerShell modules and think it would add value for cross platform tasks.

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
# PR Summary

Fixes #2290.

Added `Get-FileEncoding` cmdlet to detect encoding from file using [`System.IO.StreamReader.CurrentEncoding`](https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.currentencoding?view=net-7.0).

Also moved the `GetPathEncoding` method to `PathUtils` so it can be shared easily between cmdlets.

### Parameter Sets

```powershell
Get-FileEncoding [-Path] <string> [<CommonParameters>]

Get-FileEncoding [-LiteralPath] <string> [<CommonParameters>]
```

### Usage

```powershell
> Get-FileEncoding -Path .\.markdownlint.json

Preamble          :
BodyName          : utf-8
EncodingName      : Unicode (UTF-8)
HeaderName        : utf-8
WebName           : utf-8
WindowsCodePage   : 1200
IsBrowserDisplay  : True
IsBrowserSave     : True
IsMailNewsDisplay : True
IsMailNewsSave    : True
IsSingleByte      : False
EncoderFallback   : System.Text.EncoderReplacementFallback
DecoderFallback   : System.Text.DecoderReplacementFallback
IsReadOnly        : True
CodePage          : 65001
```

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by ThomasNieto** [comment]
The issue predates the [TextUtility](https://github.com/PowerShell/TextUtility) module. I feel this cmdlet should be in that module so earlier PowerShell versions can make use of it.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetFileEncodingCommand.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements the Get-FileEncoding command.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "FileEncoding", DefaultParameterSetName = PathParameterSet, HelpUri = "https://go.microsoft.com/fwlink/?LinkID=1234567")]
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetFileEncodingCommand.cs:12]: We need new link.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetFileEncodingCommand.cs (hunk 1 above)
Comment (ArmaanMcleod) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetFileEncodingCommand.cs:13]: Yeah put this one as a placeholder. I will also fill out documentation issue.

**Comment by SteveL-MSFT** [comment]
@PowerShell/wg-powershell-cmdlets discussed this and we agreed that this is a useful cmdlet, however, it should be part of the https://github.com/powershell/textutility module and not part of PS7 itself.

**Comment by ArmaanMcleod** [comment]
Thanks @SteveL-MSFT & @ThomasNieto for the feedback. I'll make a PR in the TextUtility repo and close this PR.

**Pull request closed without merging**